frontend/components/header.js

```diff
@@ -16,18 +16,39 @@ class Header extends HTMLElement {
   connectedCallback() {
     const rawUsername = localStorage.getItem('signedIn') === "true" ? localStorage.getItem('username') : "there";
     const username = escapeHTML(rawUsername);
-
+    const profilePicture = username === 'there' ? `assets/images/default-avatar.jpeg` : `https://github.com/${username}.png`;
+    const links = localStorage.getItem('signedIn') === "true" 
+                ? ` <a href="leaderboard.html">leaderboard</a>
```

Comment:
Just a UI suggestion, could we use icons for the other links as darkmode has an icon. for the sake of consistency in the menu. We can do it in another PR though if not possible in this one

Reply:
let's leave it for another PR

Reply:
I'll do it after the conversion